doc/specs/#500 - Auto generate manifests for newer versions.md

```diff
@@ -0,0 +1,87 @@
+
+## Problem Statement
+
+Currently, winget-pkgs relies on community contributors to update package versions, leading to delays between a new software release and its availability in WinGet. This delay is problematic when updates address security vulnerabilities, leaving users exposed. Faster updates would make WinGet more reliable, particularly for businesses that depend on timely and verified application availability.
+
+## Solution Design
+
+The proposed automation will automatically open pull requests (PRs) for new versions of applications when a new release is detected via the software’s API.
+
+```jsonc
+{
+  "$schema": "<link to schema>",
+  "PackageIdentifier": "AdGuard.AdGuard",
+  "Source": [
+    {
+      "InvokeType": "RestMethod",
+      "Uri": "",
+      "Method": "Get",
+      // other switches and args supported by Invoke-RestMethod/WebRequestHeaders
+      // like Body, UserAgent, MaximumRedirection, SkipHttpErrorCheck, ErrorAction
+    }
+  ],
+  "PostResponseScript": "$Response = $Response.response",
+  "Update": {
+    "PackageVersion": "$Response.version.'#cdata-section'",
+    "InstallerUrls": "$Response.'update-url'.'#cdata-section'"
+  },
+  "AdditionalInfo": {
+    "ForceUpgrade": false
+  },
+  "Skip": {
+    "Skip?": false,
+    "Reason": null
+  }
+}
+```
+
+The above JSON defines automation metadata and will be stored in a `.auto_updates.json` file alongside `.validation` files.
+
+### Automation Workflow
+
+The automation consists of a PowerShell script that:
+
+1. Loads variables specified in the `AdditionalInfo` object.
+2. Executes `Invoke-RestMethod` or `Invoke-WebRequest` with the defined arguments (supporting multiple sources for different APIs).
+3. Runs the `PostResponseScript` to process API responses.
+4. Calls `wingetcreate update` in non-interactive mode:
+   - Uses `--submit` if no additional metadata is required.
+   - If additional metadata is needed, modifies the manifests manually before calling `wingetcreate submit`.
+
+This PowerShell script will be executed in an Azure Pipeline on a set schedule.
+
+## UI/UX Design
+
+N/A. This change does not impact the WinGet CLI directly.
+
+## Security
+
+Security remains unaffected. PRs generated by the automation will go through the same validation pipelines and require manual moderator approval before merging.
+
+## Compatibility
+
+No breaking changes. This automation can coexist with the existing @wingetbot installer hash updates.
+
+## Potential Issues
+
+- **GitHub API Rate Limits**: Applications distributed via GitHub Releases may face API rate limits. Encouraging project maintainers to directly update WinGet manifests (push-style automation) could help mitigate this issue.
```

Comment:
Not just GitHub API Rate Limits, but what about other sites that have rate limits such as Mozilla? If they get hit with a few hundred requests quickly enough, they could throttle the connections or toss back 403 or 429 error